sys/kern/lockdep.c

```diff
@@ -6,21 +6,27 @@
 #include <sys/mimiker.h>
 #include <sys/klog.h>
 
-typedef TAILQ_HEAD(, lock_class) lock_class_list_t;
-
 typedef struct lock_class {
-  TAILQ_ENTRY(lock_class) hash_entry;
+  SIMPLEQ_ENTRY(lock_class) hash_entry;
   lock_class_key_t *key;
   const char *name;
-  TAILQ_HEAD(, lock_class_link) locked_after;
 
-  int bfs_visited;
+  /* A list of lock classes corresponding to locks acquired while holding a lock
+   * belonging to this lock class. */
+  SIMPLEQ_HEAD(, lock_class_link) locked_after;
+
+  /* Used in the graph traversal to decide if the graph node was already
+   * visited. */
+  int bfs_gen_id;
 } lock_class_t;
 
-/* lock_class_link represents an edge in the graph of lock depedency. */
+/*
+ * lock_class_link represents an edge in the graph of lock depedency. An edge
+ * C1-> C2 is created if the lock class C1 acquired while holding the class C2.
```

Comment:
I think it's the other way around: An edge C1 -> C2 is created if a lock inside lock class C2 is acquired while holding a lock belonging to lock class C1.